VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/SparQLWhere.cs

```diff
@@ -59,64 +59,18 @@ public SparQLWhere SubjectIs(string subject)
         /// input parameter.
         /// </summary>
         /// <param name="predicate">Predicate of the sparql triplet</param>
-        /// <returns></returns>
+        /// <returns>this</returns>
         public SparQLWhere PredicateIs(string predicate)
         {
-            if (EncodedSPOs.ContainsKey(predicate))
-            {
-                PredicateString = EncodedSPOs[predicate].Name;
-            }
-            else
-            {
-                PredicateString = predicate;
-            }
-            if (TripletIsDone())
-            {
-                AddCondition();
-            }
-            return this;
-        }
-
-        /// <summary>
-        /// Specifies the object of the sparql triplet as being equal to the
-        /// input parameter.
-        /// </summary>
-        /// <param name="obj">Object of the sparql triplet</param>
-        /// <returns></returns>
-        public SparQLWhere ObjectIs(string obj)
-        {
-            ObjectString = obj;
-            if (TripletIsDone())
-            {
-                AddCondition();
-            }
-            return this;
-        }
+            if (PredicateString != "") throw new ArgumentException("Predicate in query has already been set. Finish current condition first");
 
-        /// <summary>
-        /// Specifies the name of the variable
-        /// </summary>
-        /// <param name="subject">name of the subject variable</param>
-        /// <returns></returns>
-        public SparQLWhere SubjectAs(string subject)
-        {
-            SubjectString = $"?{subject}";
-            if (TripletIsDone())
+            if (!EncodedSPOs.ContainsKey(predicate))
             {
-                AddCondition();
+                throw new KeyNotFoundException("Encoded SPO does not include the predicate key");
             }
```

Comment:
Hvis det første guard clause er single line tænker jeg også det her skal være det, eller bliver det for bredt?